File: src/B_2798/B_2798.java
package B_2798;

import java.io.*;
import java.util.StringTokenizer;

public class B_2798 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        st=new StringTokenizer(br.readLine()," ");
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[] arr = new int[N];
        st = new StringTokenizer(br.readLine()," ");
        for(int i=0; i<N; i++){
            arr[i]=Integer.parseInt(st.nextToken());
        }
        int res=0;
        for(int i=0; i<N-2; i++){
            for(int k=i+1; k<N-1; k++){
                for(int m=k+1; m<N; m++){
                    int temp = arr[i]+arr[k]+arr[m];
                    if(temp == M){
                        System.out.println(temp);
                        return;
                    }
                    if (temp > res && temp < M) {
                        res = temp;
                    }
                }
            }
        }
        System.out.println(res);
    }
}

File: src/B_2798/B_2798.kt
package B_2798

import java.io.*
import java.util.*

fun main(args: Array<String>){
    val br = BufferedReader(InputStreamReader(System.`in`))
    var st: StringTokenizer
    st = StringTokenizer(br.readLine(), " ")
    val N = st.nextToken().toInt()
    val M = st.nextToken().toInt()
    val arr = IntArray(N)
    st = StringTokenizer(br.readLine(), " ")
    for (i in 0 until N) {
        arr[i] = st.nextToken().toInt()
    }
    var res:Int = 0
    for(i in 0 until N-2){
        for(k in i+1 until N-1){
            for(m in k+1 until N){
                var temp = arr[i]+arr[k]+arr[m]

                if(M == temp){
                    print(temp)
                    return
                }
                if(temp < M && temp > res){
                    res = temp
                }
            }
        }
    }
    print(res)
}